src/Reveal.Sdk.Dom.Tests/Visualizations/VisualizationFixture.cs

```diff
@@ -0,0 +1,445 @@
+﻿using Xunit;
+using Moq;
+using Reveal.Sdk.Dom.Data;
+using Reveal.Sdk.Dom.Visualizations;
+using System.Collections.Generic;
+using Xunit.Sdk;
+using Newtonsoft.Json.Linq;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Reveal.Sdk.Dom.Core.Utilities;
+using System;
+
+namespace Reveal.Sdk.Dom.Tests.Visualizations
+{
+    public class VisualizationFixture
+    {
+        [Fact]
+        public void Constructor_SetTitleAndChartType_WhenConstructed()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem { DataSource = dataSource };
+            var testTitle = "testTitle";
+
+            // Act
+            var visualization = new Mock<Visualization>(testTitle, dataSourceItem) { CallBase = true }.Object;
+
+            // Assert
+            Assert.Equal(testTitle, visualization.Title);
+            Assert.Equal(ChartType.Unknown, visualization.ChartType);
+        }
+
+
+        [Fact]
+        public void Constructor_UseTabularDataDefinition_WhenUseTabularDataSource()
```

Comment:
We can combine the unit tests for the case when the field `HasTabularData` is NOT defined/ set as true or false, by using `Theory`

Reply:
Actually, I think they are very different, and Xmla case will be changed in the future. So I think it's better to have distinguish between them